### DISCUSSION TITLE
Best way to model self-shade loss in c-Si with pvlib?

### DISCUSSION THREAD

**Discussion by williamhobbs** [pvlib/pvlib-python#1880] [Q&A] [answered]
I'm interested in modeling row to row self-shade losses in large crystalline silicon plants. I have two specific use cases in mind:
1. Fixed-tilt systems
2. Tracking systems with tracking problems (e.g., tracker clocks have drifted by 30 minutes)

**What is the best way to do this currently in pvlib?**

I'm thinking about using `infinite_sheds` to calculate diffuse fraction `fd` and shaded fraction `fs`, using https://github.com/pvlib/pvlib-python/pull/1871, and then use one of these approaches to calculate shade losses:
1. use @kandersolar's simple piece-wise shading model from https://www.nrel.gov/docs/fy20osti/76023.pdf (see also https://groups.google.com/g/pvlib-python/c/ru1KCnz_hkU/m/NslEU1nTDwAJ) 
2. create a _generic_ normalized power-`fd`-`fs` table like in https://pvlib-python.readthedocs.io/en/stable/gallery/shading/plot_partial_module_shading_simple.html. I could create the table once, and then use `scipy.interpolate.LinearNDInterpolator` to interpolate for a given `fs` and `fd` in each timestep
3. same as option 2, but specific to the module parameters for the site (e.g., use the SAM CEC database)

For option 3, I'm not sure how much difference it makes vs option 2, or where to get breakdown values needed for `singlediode.bishop88`.

**Comment by cwhanse** [comment, marked as answer]
If you are mostly interested in DC power/energy losses, and you assume that strings stay near their maximum power points, then I don't think that #3 differs meaningfully from #2. If you also want accurate calculation of Isc, then #3 becomes relevant.

**Reply by williamhobbs** [reply]
Thanks. It is power/energy that I’m interested in. 

Regarding max power point, that is my assumption for tracking systems, since they are all 1P configurations and all modules would see the same shade (with the simplifying assumption that everything is perfectly aligned…). 

For fixed tilt, many plants are 2P with central inverters, where strings on the top are in parallel with strings on the bottom. I’d have to work out `fs`, `fd`, and shade losses for top and bottom rows separately, but then I was planning to assume that there were no voltage mismatch losses. Do you think that is compatible with assuming that “strings stay near their maximum power points”?

**Reply by cwhanse** [reply]
Voltage mismatch in series from variation in irradiance isn't significant, if one continues to assume a uniform temperature. Voltage differences between parallel strings are handled by the inverter at its separate MPPT inputs, if the strings are separated. If instead the strings are simply joined at a common bus: the inverter will still set the common bus voltage, but it should (will?) choose the voltage based on the string that gives the greatest power, and the other string will shift its power point to conform. There no function in pvlib for this parallel-strings case.

**Reply by williamhobbs** [reply]
Ok, that agrees with what I was assuming. With central inverters (and some string inverters, I think), the top and bottom strings would be joined at a common bus. 

Your point about inverter MPPT picking a voltage that favors the higher power string(s) is a good one. It means, among other things, that getting temperature correct for the partially shaded strings matters less, since they have less impact on total power.

**Reply by cwhanse** [reply]
I wrote "it should (will?) choose the voltage based on the string that gives the greatest power, and the other string will shift its power point to conform"

This is the ideal case for 2 parallel strings. For more than 2 parallel strings that are joined at a common bus, the ideal case is that the inverter finds the voltage for the maximum power of the summed (voltage, power) curves from each string. That may not be at the same voltage as the maximum power for any individual string (although I think in the case of parallel, partially shaded strings, it will often be at the maximum power point of the string with the greatest power).

**Comment by williamhobbs** [comment]
I've tested out options #&#x2060;1 and #&#x2060;2 and compared with a 2P fixed tilt plant on a (relatively) clear winter day. The simple shade model looks like a pretty good match!

**[2024-01-29 edit: I think the GCR calculations I used are incorrect]**

![shaded_power_comparison](https://github.com/pvlib/pvlib-python/assets/45701090/944ca57c-91f7-437a-8047-f9e4417a63d7)

The plant has an azimuth >180, which is why the profile is asymmetrical. (observed power was linearly scaled to match its peak with that of the modeled shaded power)

**Reply by williamhobbs** [reply]
**[2024-01-29 edit: I think the GCR calculations I used are incorrect]**

Here's my code (zipped jupyter notebook), in case anyone is interested in details:

[Self-shade_examples_rev4b.zip](https://github.com/pvlib/pvlib-python/files/12768510/Self-shade_examples_rev4b.zip)

**Reply by mikofski** [reply]
Nice!
